pkg/util/slidingwindow/helpers.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package slidingwindow
```

Comment:
Perhaps the obs-infers on this review have opinions.

Reply:
I'm planning on using this pkg within the simulator for replicastats - since that has a lot of additional logic and locking which would be nicely abstracted here.